component/attr_cache/attr_cache.go

```diff
@@ -566,15 +512,25 @@ func (ac *AttrCache) IsDirEmpty(options internal.IsDirEmptyOptions) bool {
 }
 
 func (ac *AttrCache) anyContentsInCache(prefix string) bool {
-	// Add a trailing / so that we only find child paths under the directory and not paths that have the same prefix
-	prefix = dirToPrefix(prefix)
 	ac.cacheLock.RLock()
 	defer ac.cacheLock.RUnlock()
-	for key, value := range ac.cacheMap {
-		if strings.HasPrefix(key, prefix) && value.valid() && value.exists() {
+
+	cachedContentItem, err := ac.cacheMap.get(prefix)
```

Comment:
`cachedContentItem` is a confusing variable name. Maybe `directory` would be better?